lib/ProducerImpl.cc

```diff
@@ -912,10 +912,9 @@ bool ProducerImpl::ackReceived(uint64_t sequenceId, MessageId& rawMessageId) {
 
     if (op.chunkedMessageId) {
         // Handling the chunk message id.
-        if (op.chunkId == 0) {
-            op.chunkedMessageId->setFirstChunkMessageId(messageId);
-        } else if (op.chunkId == op.numChunks - 1) {
-            op.chunkedMessageId->setLastChunkMessageId(messageId);
+        op.chunkMessageIdList.push_back(messageId);
+        if (op.chunkId == op.numChunks - 1) {
+            op.chunkedMessageId->setChunkedMessageIds(std::move(op.chunkMessageIdList));
             messageId = op.chunkedMessageId->build();
```

Comment:
BTW, this logic seems wrong? We create `OpSendMsgs` for each single chunk. It seems that `op.chunkMessageIdList` still has 1 item.